staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1006,7 +1006,7 @@ func (c *cacheWatcher) sendWatchCacheEvent(event *watchCacheEvent) {
 	}
 }
 
-func (c *cacheWatcher) process(initEvents []*watchCacheEvent, resourceVersion uint64) {
+func (c *cacheWatcher) process(initEvents []*watchCacheEvent, resourceVersion uint64, timeout time.Duration) {
```

Comment:
timer shoud be started at the started at the beginning of this function, not after sending all initial events.